app/containers/ExplorePage.tsx

```diff
@@ -140,12 +146,43 @@ export default function ExplorePage() {
   };
 
   const handlePredictionOverride = (location: string, override: CreatableOption | null) => {
+    if (observations === undefined) return false;
     const overrides = { ...predictionOverrides };
+    const observationIndex: number = observations.findIndex(obs => obs.location === location);
+    const observation = observations[observationIndex];
+
     if (override === null) {
       delete overrides[location];
     } else {
       overrides[location] = override;
     }
+    observations[observationIndex] = {
+      ...observation,
+      label: override === null ? observation.pred_1 : override.value
+    };
+    setPredictionOverrides(overrides);
+    return false;
+  };
+
+  const detectOverrides = async (dataObservations: Observation[] | undefined) => {
```

Comment:
This function doesn't need to be `async`.

Reply:
fixed